File: 프로그래머스/lv0/120923. 연속된 수의 합/연속된 수의 합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int num, int total)
{
    vector<int> answer;
    int a = 0;
    int sum2 = 0;
    int count = -100;
    while (1)
    {   
        vector<int> temp;
        int sum = 0;
        a = count;
        for (int i = 0; i < num; i++)
        {
            sum = sum + a;
            temp.push_back(a);
            a++;
        }
        if (sum == total)
        {
            sum2 = sum;
            answer = temp;
            break;
        }
        count++;
    }

    return answer;
}

File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.java
import java.util.Arrays;
import java.util.stream.IntStream;

class Solution {

    public int[] solution(int num, int total) {
        int sum = -1;
        int idx = -100;
        while (sum != total) {
            sum = 0;
            for (int i = idx; i < idx + num; i++) {
                sum += i;
            }
            idx++;
        }

        return IntStream.range(idx - 1, idx + num - 1).toArray();
    }
}